### PR TITLE
docs: Fix broken links with sphinx linkchecker

### PR DESCRIPTION
**Pull request by bruno-at-bareos**
Documentation use sphinx linkcheck to fix broken links

- Adding linkcheck options to conf.py
- Fix all broken link in documentation

 Sphinx linkcheck can be used during documentation build

```
sphinx-build -M linkcheck docs/manuals/source docs/BareosMainReference -j2
```

The automatic usage during documentation build, still need to be discussed.

#### Please check

- [x] Short description and the purpose of this PR is present _above this paragraph_
- [x] Your name is present in the AUTHORS file (optional)

If you have any questions or problems, please give a comment in the PR.

### Helpful documentation and best practices

- [Git Workflow](https://docs.bareos.org/DeveloperGuide/gitworkflow.html)
- [Automatic Sourcecode Formatting](https://docs.bareos.org/DeveloperGuide/generaldevel.html#automatic-sourcecode-formatting)
- [Check your commit messages](https://docs.bareos.org/DeveloperGuide/gitworkflow.html#commits)

##### General
- [x] PR name is meaningful
- [x] Purpose of the PR is understood
- [x] Separate commit for this PR in the CHANGELOG.md, PR number referenced is same
- [x] Commit descriptions are understandable and well formatted
- ~~[ ] If backport: add original PR number and target branch at top of this file: **Backport of PR#000 to bareos-2x**~~

##### Source code quality

- [x] Source code changes are understandable
- [x] Variable and function names are meaningful
- [x] Code comments are correct (logically and spelling)
- [x] Required documentation changes are present and part of the PR
- [x] `bareos-check-sources --since-merge` does not report any problems
- [x] `git status` should not report modifications in the source tree after building and testing

### CODE REVIEW

**Review by joergsteffens — CHANGES_REQUESTED** [review]
I'm very glad you found and configured this tool.
I've suggestions on some of the internal URLs, but besides that, I'm fine with this PR.

File: docs/manuals/source/DeveloperGuide/generaldevel.rst [hunk 1]
```diff
@@ -45,7 +45,7 @@ Also, include the Bareos copyright notice that can be found in every source file
 Bug Database
 ------------
 
-We have a bug database which is at https://bugs.bareos.org.
+We have a bug database which is at https://bugs.bareos.org/my_view_page.php.
```
[the remark this answers is not in the document]
Comment (bruno-at-bareos) [reply docs/manuals/source/DeveloperGuide/generaldevel.rst]: Thanks for the pointer was a left over from applying all redirections, but we skip now bareos domain.

File: docs/manuals/source/TasksAndConcepts/TransportEncryption.rst [hunk 2]
```diff
@@ -78,7 +78,7 @@ To get a trusted certificate (CA or Certificate Authority signed certificate), y
 
 Bareos is known to work well with RSA certificates.
 
-You can use programs like `xca <http://xca.sourceforge.net/>`_ or TinyCA to easily manage your own CA with a Graphical User Interface.
+You can use programs like `xca <https://www.hohnstaedt.de/xca/>`_ or TinyCA to easily manage your own CA with a Graphical User Interface.
```
[the remark this answers is not in the document]
Comment (bruno-at-bareos) [reply docs/manuals/source/TasksAndConcepts/TransportEncryption.rst:81]: Thanks nice suggestion.

File: docs/manuals/source/TasksAndConcepts/AutochangerSupport.rst [hunk 3]
```diff
@@ -505,7 +505,7 @@ Tapespeed and blocksizes
 :index:`\ <single: Tape; speed>`
 :index:`\ <single: Blocksize; optimize>`
 
-The `Bareos Whitepaper Tape Speed Tuning <http://www.bareos.org/en/Whitepapers/articles/Speed_Tuning_of_Tape_Drives.html>`_ shows that the two parameters :strong:`Maximum File Size`\  and :strong:`Maximum Block Size`\  of the device have significant influence on the tape speed.
+The `Bareos Whitepaper Tape Speed Tuning <https://www.bareos.com/wp-content/uploads/2021/05/optimizing-the-tape-speed.pdf>`_ shows that the two parameters :strong:`Maximum File Size`\  and :strong:`Maximum Block Size`\  of the device have significant influence on the tape speed.
```
[the remark this answers is not in the document]
Comment (bruno-at-bareos) [reply docs/manuals/source/TasksAndConcepts/AutochangerSupport.rst:508]: Nice, a lot better.

**Review by joergsteffens — APPROVED** [review]
Very nice!

**Pull request merged**